pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractDispatcherSingleActiveConsumer.java

```diff
@@ -166,6 +166,10 @@ public synchronized CompletableFuture<Void> addConsumer(Consumer consumer) {
         }
 
         if (subscriptionType == SubType.Exclusive && !consumers.isEmpty()) {
+            Consumer actConsumer = ACTIVE_CONSUMER_UPDATER.get(this);
+            if (actConsumer != null) {
+                actConsumer.cnx().checkConnectionLiveness();
+            }
```

Comment:
But the new consumer will not retry to connect to the topic, right? Do we need to wait for the connection liveness check done?

Reply:
@codelipenghui 

> But the new consumer will not retry to connect to the topic, right? Do we need to wait for the connection liveness check to be done?

Sure, I did this improve.

I have a concern: 

**Background**: the PR https://github.com/apache/pulsar/pull/20026 changed the method `dispatcher.addConsumer` to an asynchronous method, it broke the lock of `synchronized(dispathcer.this)`, this change only affected the releases larger than `3.0.0`.

**Concern**: The improvement "wait for the connection liveness check done" relies on the asynchronous method `dispatcher.addConsumer`. I am thinking about whether to accept the patch https://github.com/apache/pulsar/pull/20026 and fix the broken lock, or revert this patch to make the logic simple.

Reply:
After talking with @codelipenghui @BewareMyPower @RobertIndie @gaoran10, I will write a new PR to fix the lock that broke by the PR https://github.com/apache/pulsar/pull/20026